### PR TITLE
refactor(jkube-kit/common): Extract errorStream processing in ExternalCommand enabling overrides

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

Related to https://github.com/eclipse/jkube/issues/2453

In order to read command's errorStream during failure, we need to refactor ExternalCommand to expose this logic so that child classes can override this logic as per the default behavior (i.e logging it as a warn message)

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] My code follows the style guidelines of this project
 - [X] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [X] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [X] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [X] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ExternalCommandTest.java [hunk 1]
```diff
@@ -47,6 +55,30 @@ void execute_whenCommandFailed_thenThrowException() {
...
+    assertThatIOException()
+        .isThrownBy(testCommand::execute)
+        .withMessage("Process 'java idontexist' exited with status 1");
+    verify(kitLogger).warn("Error: Could not find or load main class idontexist");
+  }
+
+  @Test
+  void execute_whenCommandFailedButProcessErrorOverridden_thenErrorNotBlank() {
+    // Given
+    TestCommandWithOutputError testCommand = new TestCommandWithOutputError(kitLogger, new String[] {"java", "idontexist"}, null);
```
[the remark this answers is not in the document]
Comment (rohanKanojia) [reply jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ExternalCommandTest.java:73]: You're right. I'm updating test to use inline method.

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ExternalCommandTest.java [hunk 2]
```diff
@@ -13,21 +13,29 @@
...
+  @BeforeEach
+  void setUp() {
+    kitLogger = spy(new KitLogger.SilentLogger());
+  }
 
   @Test
   void execute_whenCommandCompletedSuccessfully_thenPrintResult() throws IOException {
     // Given
-    TestCommand testCommand = new TestCommand(kitLogger, new String[] {"echo", "hello"});
+    TestCommandWithOutputError testCommand = new TestCommandWithOutputError(kitLogger, new String[] {"echo", "hello"}, null);
```
[the remark this answers is not in the document]
Comment (rohanKanojia) [reply jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ExternalCommandTest.java:38]: Okay, you're right. Let me revert these changes

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ExternalCommandTest.java [hunk 3]
```diff
@@ -13,21 +13,29 @@
...
+  @BeforeEach
+  void setUp() {
+    kitLogger = spy(new KitLogger.SilentLogger());
+  }
 
   @Test
   void execute_whenCommandCompletedSuccessfully_thenPrintResult() throws IOException {
     // Given
-    TestCommand testCommand = new TestCommand(kitLogger, new String[] {"echo", "hello"});
+    TestCommand testCommand = new TestCommand(kitLogger, new String[] {"echo", "hello"}, null);
```
Comment (sunix) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ExternalCommandTest.java:38]: is null required?

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ExternalCommandTest.java [hunk 4]
```diff
@@ -47,6 +55,36 @@ void execute_whenCommandFailed_thenThrowException() {
...
+        .isThrownBy(testCommand::execute)
+        .withMessage("Process 'java idontexist' exited with status 1");
+    verify(kitLogger).warn("Error: Could not find or load main class idontexist");
+  }
+
+  @Test
+  void execute_whenCommandFailedButProcessErrorOverridden_thenErrorNotBlank() {
+    // Given
+    final StringBuilder errorBuilder = new StringBuilder();
+    TestCommand testCommand = new TestCommand(kitLogger, new String[] {"java", "idontexist"}, null) {
```
Comment (sunix) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ExternalCommandTest.java:74]: maybe rename that variable name:
overriddenProcessErrorTestCommand maybe

**Review by sunix — APPROVED** [review]

**Review by manusa — APPROVED** [review]

**Pull request merged**